plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -1179,22 +1179,64 @@ def _check_packages(existing_packages, new_packages):
             return existing_packages == new_packages
         return False
 
+    def create_legacy_metadata_dict(self):
+        """
+        Generate metadata dict and add erratum pkglist to it since it is stored
+        in a separate MongoDB collection.
+
+        This dict is used during incremetal export of erratum in PublishErrataStepIncremental.
+        """
+        metadata_dict = super(Errata, self).create_legacy_metadata_dict()
+        pkglists = models.ErratumPkglist.objects(errata_id=erratum_unit.errata_id)
+        metadata_dict['pkglist'] = [pkglist['collections'] for pkglist in pkglists]
+        return metadata_dict
+
+    def import_erratum(self, repo_id):
+        """
+        To avoid importing erratum units without pkglist:
+        - merge and save pkglist to its own collection
+        - then save erratum unit itself
```

Comment:
The problem is what to do first. Save erratum itself and then its pkglist or vice versa.
I think it is safer to have pkglists first and then save erratum - in this case we won't have "broken" errata but we can have pkglists which has no use and won't be cleaned up ever(?). Lesser of two evils. Thoughts?